Rubberduck.Main/Root/RubberduckIoCInstaller.cs

```diff
@@ -200,33 +205,35 @@ private void RegisterFactories(IWindsorContainer container, Assembly[] assemblie
         {
             foreach (var assembly in assembliesToRegister)
             {
-                var assemblyTypes = GetIoCRegisteredTypes(assembly);
-
-                container.Register(Types.From(assemblyTypes)
-                    .Where(type => type.IsInterface && type.Name.EndsWith("Factory"))
+                container.Register(Types.FromAssembly(assembly)
+                    .IncludeNonPublicTypes()
+                    .Where(type => type.IsInterface && type.Name.EndsWith("Factory") && type.NotDisabledExperimental(_initialSettings))
+                    .WithService.Self()
                     .Configure(c => c.AsFactory())
                     .LifestyleSingleton());
             }
         }
 
         private static void RegisterSpecialFactories(IWindsorContainer container)
         {
-            container.Register(Component.For<IFolderBrowserFactory>()
-                .ImplementedBy<DialogFactory>()
-                .LifestyleSingleton());
+            //container.Register(Component.For<IDeclarationFinderFactory>()
+            //    .ImplementedBy<DeclarationFinderFactory>()
+            //    .LifestyleSingleton());
             container.Register(Component.For<IFakesProviderFactory>()
                 .ImplementedBy<FakesProviderFactory>()
                 .LifestyleSingleton());
+            container.Register(Component.For<IFolderBrowserFactory>()
+                .ImplementedBy<DialogFactory>()
+                .LifestyleSingleton());
         }
 
         private void RegisterQuickFixes(IWindsorContainer container, Assembly[] assembliesToRegister)
         {
             foreach (var assembly in assembliesToRegister)
             {
-                var assemblyTypes = GetIoCRegisteredTypes(assembly);
-
-                container.Register(Classes.From(assemblyTypes)
-                    .BasedOn<IQuickFix>()
+                container.Register(Classes.FromAssembly(assembly)
+                    .IncludeNonPublicTypes()
+                    .Where(type => type.IsBasedOn(typeof(IQuickFix)) && type.NotDisabledExperimental(_initialSettings))
```

Comment:
This will not work, because, as commented in the `TypeExtensions` class, the implementation of `IsBasedOn` checks something a lot more restrictive then CW`s `BasedOn`.

Reply:
fixed